### ISSUE TITLE
iOS banner and modal for unconfirmed users

### ISSUE THREAD

**Issue by loarie** [inaturalist/INaturalistIOS#687]
For any unconfirmed user on any iNaturalist app, add a green banner at the top of the site styled like this:
<img width="288" alt="Screenshot 2024-03-04 at 1 58 25 PM" src="https://github.com/inaturalist/INaturalistIOS/assets/598026/35f405c1-212c-4462-a755-fd3a93486cd7">
The banner should say:
"Please confirm that [email address] is the best way to contact you. Click here to resend a confirmation email"

When the user clicks on the underlined potion reveal a modal that says:
"Please click the link sent to [email address] to confirm your email address."

**Comment by kueda** [comment]
Same comment as I [made](https://github.com/inaturalist/iNaturalistAndroid/issues/1335#issuecomment-1979919975) on the Android equivalent:

> If @loarie wants the user to be able to resend the confirmation email, you're going to have to use the [undocumented](https://github.com/inaturalist/iNaturalistAPI/issues/427) `/users/resend_confirmation` endpoint, which requires auth and resends the email to the authenticated user.
>
> FWIW, I think it's very confusing that the banner doesn't match the screenshot, and that the banner text implies that an email will be sent but tapping the banner just opens a modal that does nothing.

**Comment by alexshepard** [comment]
Is there a reason this can't just be a notification?

**Comment by loarie** [comment]
@abhasinat can you chime in on the UX here? (I think his intent was to make it more annoying)

**Comment by abhasinat** [comment]
Yep, the intent was to make it more pervasive and annoying to encourage a user to confirm their email. We also decided to change the text to indicate that a confirmation email would be resent (see below). Clicking the banner should open a modal with the text: "Would you like to resend a confirmation email to [email address]?" with the buttons "Resend Email" and "Cancel"

After an email is sent, there should be a modal with the text "Email sent! Please click the link in the email." with a single CTA "Ok"

<img width="960" alt="Mobile Announcement - Confirmation" src="https://github.com/inaturalist/INaturalistIOS/assets/44380775/5cabe5fe-9a0a-4ca4-8d12-305b31156ce5">

**Comment by alexshepard** [comment]
ok, Tony and I can talk about this tomorrow at our iOS app sync. when do you want this by?

is it ok if this replaces the notification slot on the phone? or, if there is both this "banner" and a notification active, what do you expect the user to see?

**Comment by abhasinat** [comment]
By notification, I assume you mean a mobile announcement? if there is a mobile announcement that is dismissible, I expect a user to only see that announcement. once they dismiss it, they should see this persistent banner. Basically, I only want one banner or announcement to show at a time, and I think the non-persistent one should take priority. If @loarie feels differently, I defer to him. scott will also be a better person to answer about timeline.

**Comment by loarie** [comment]
Abhas's UX plan sound good to me

RE: timeline, this is done on the website and android so while there's no hard deadline, would be great to get this wrapped up within the next few weeks.
